server/tournament.py

```diff
@@ -544,6 +576,50 @@ def update_tournament_spirit_rankings(tournament: Tournament) -> None:
     tournament.save()
 
 
+def check_if_authorised_to_submit_scores_by_match(match: Match, player: Player) -> tuple[bool, int]:
+    registrations = UCRegistration.objects.filter(
+        event=match.tournament.event, person_id=player.ultimate_central_id
+    )
+
+    for reg in registrations:
+        if (
+            match.team_1 is not None
+            and match.team_2 is not None
+            and reg.team.id in {match.team_1.id, match.team_2.id}
+            and any(role in reg.roles for role in ROLES_ELIGIBLE_TO_SUBMIT_SCORES)
+        ):
+            return True, reg.team.id
+
+    return False, 0
+
+
+def check_if_authorised_to_submit_scores_by_tournament(
```

Comment:
Can we rename this to something like `can_submit_tournament_scores`?

Reply:
yep